pkg/utils/filters/rewrite.go

```diff
@@ -0,0 +1,40 @@
+package filters
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const OriginPathHeaderKey = "X-Rewrite-Original-Path"
+const OldResourceApiServerPrefix = "/apis/clusterpedia.io/v1beta1/resources"
+
+func WithRewriteFilter(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if !doUrlRewrite(req) {
+			klog.V(5).InfoS("request not need rewrite", "path", req.URL.EscapedPath())
+		}
+
+		handler.ServeHTTP(w, req)
+	})
+}
+
+func doUrlRewrite(req *http.Request) bool {
+	oldPath := req.URL.EscapedPath()
+	if strings.HasPrefix(oldPath, OldResourceApiServerPrefix) {
+		return false
+	}
+
+	rewritePath, err := url.JoinPath(OldResourceApiServerPrefix, req.URL.Path)
+	if err != nil {
+		return false
+	}
+	req.URL.Path = rewritePath
+	req.Header.Set(OriginPathHeaderKey, oldPath)
+
+	klog.V(5).InfoS("Rewrite url", "oldPath", oldPath, "newPath", req.URL.EscapedPath())
+
+	return true
+}
```

Comment:
I thinks we just need to cover this function in unit tests without starting the http server

```suggestion
func urlPrefixRewrite(oldPath string) (string, bool) {
	if strings.HasPrefix(oldPath, OldResourceApiServerPrefix) {
		return "", false
	}

	rewritePath, err := url.JoinPath(OldResourceApiServerPrefix, req.URL.Path)
	if err != nil {
		return "", false
	}

	return rewritePath, true
}
```